### PR TITLE
[SR-1608] NSStream unimplemented features PT1

### PR DESCRIPTION
**Pull request by DaveLiu888**
Old pull: https://github.com/apple/swift-corelibs-foundation/pull/489
Old Pull request was closed due to churn about experimental un official API particularly 

```
class func getStreamsToHost(withName hostname: String, port: Int, inputStream: AutoreleasingUnsafeMutablePointer<InputStream?>?, outputStream: AutoreleasingUnsafeMutablePointer<NSOutputStream?>?) 
```

I have decoupled these 2 issues and created a separate Jira ticket for it here and will be filing a pr for soon https://bugs.swift.org/browse/SR-2431

**What's in this pull request?**
1st part of Resolved bug number: 1608 (https://bugs.swift.org/browse/SR-1608)
NSStream implementation of unimplemented features with tests

SR-1608 has been split in 2 pulls in effort to keep pulls small
- [ ] @phausler 
- [ ] @parkera

### CODE REVIEW

File: Foundation/NSStream.swift [hunk 1]
```diff
@@ -67,15 +67,41 @@ extension Stream {
     }
 }
 
+extension Stream {
+    internal func _handleStreamEvent(_ event:CFStreamEventType){
+        switch event {
+        case CFStreamEventType.errorOccurred:
+            delegate?.stream(self, handleEvent: Stream.Event.errorOccurred)
+            break
```
Comment (xwu) [inline Foundation/NSStream.swift]: Is there a reason you're using explicit `break` statements?

File: Foundation/NSStream.swift [hunk 2]
```diff
@@ -155,19 +212,76 @@ open class InputStream: Stream {
...
+    open override var streamError: NSError?{
+        let error = CFReadStreamCopyError(_stream)
+        return error?._nsObject
+    }
     
     open override var streamStatus: Status {
         return Stream.Status(rawValue: UInt(CFReadStreamGetStatus(_stream)))!
     }
+
+    override open func schedule(in aRunLoop: RunLoop, forMode mode: RunLoopMode) {
```
Comment (xwu) [inline Foundation/NSStream.swift]: Nit: `open override` or `override open` -- pick one :)

File: Foundation/NSStream.swift [hunk 3]
```diff
@@ -155,19 +212,76 @@ open class InputStream: Stream {
...
     
-    private  var _stream: CFWriteStream!
+    private var _stream: CFWriteStream!
+    
+    override open var delegate: StreamDelegate?{
+        didSet{
+            switch (delegate, oldValue) {
+            case (.none , .some): _removeWriteStreamClient(); break
+            case (.some , .none): _setWriteStreamClient(); break
+            default: break;
```
Comment (xwu) [inline Foundation/NSStream.swift]: Nit: extraneous trailing semicolon

File: Foundation/NSStream.swift [hunk 4]
```diff
@@ -155,19 +212,76 @@ open class InputStream: Stream {
...
 // Currently this is left as named NSOutputStream due to conflicts with the standard library's text streaming target protocol named OutputStream (which ideally should be renamed)
 open class NSOutputStream : Stream {
     
-    private  var _stream: CFWriteStream!
+    private var _stream: CFWriteStream!
+    
+    override open var delegate: StreamDelegate?{
+        didSet{
+            switch (delegate, oldValue) {
+            case (.none , .some): _removeWriteStreamClient(); break
```
Comment (xwu) [inline Foundation/NSStream.swift]: Same question here about explicit `break`.

File: Foundation/NSStream.swift (hunk 1 above)
Comment (DaveLiu888) [inline Foundation/NSStream.swift]: no particular reason just a  habit from ObjC even though i know that swift switch don't fall through kinda like the referring to self  outside of closure habit  , thanks pointing them out removing them.

File: Foundation/NSStream.swift (hunk 2 above)
Comment (DaveLiu888) [inline Foundation/NSStream.swift]: thnx

File: Foundation/NSStream.swift (hunk 3 above)
Comment (DaveLiu888) [inline Foundation/NSStream.swift]: thnx

**Comment by alblue** [comment]
This pull request has gone stale; closing.

**Pull request closed without merging**